spyder_terminal/widgets/main_widget.py

```diff
@@ -135,11 +133,19 @@ def get_title(self):
     def setup(self):
         """Perform the setup of plugin's main menu and signals."""
         self.cmd = find_program(self.get_conf('shell'))
-        self.server = subprocess.Popen(
-            [sys.executable, '-m', 'spyder_terminal.server',
-             '--port', str(self.port), '--shell', self.cmd],
-            stdout=self.server_stdout,
-            stderr=self.server_stderr)
+        server_args = [
+            sys.executable, '-m', 'spyder_terminal.server',
+             '--port', str(self.port), '--shell', self.cmd]
+        self.server = QProcess(self)
+        env = self.server.processEnvironment()
+        for var in os.environ:
+            env.insert(var, os.environ[var])
+        self.server.setProcessEnvironment(env)
+        self.server.setProcessChannelMode(QProcess.SeparateChannels)
```

Comment:
The nice thing about using QProcess is that you can detect if the external process was not started and report it to users.

That's what we do for the Python language server in Spyder, by connecting a slot to the `errorOccurred` signal of `QProcess`:

https://github.com/spyder-ide/spyder/blob/e793c9a0f5bf95b0ce31ec724913447583f689f7/spyder/plugins/completion/providers/languageserver/client.py#L298

So you could do something similar for Spyder-terminal, by making the equivalent of `handle_process_errors` here to display a QMessageBox telling users something like this:

> The server that creates Spyder terminals failed to start. Please restart Spyder in a system terminal with the command `spyder --debug-info minimal` and look for files called `spyder_terminal_out.log` and `spyder_terminal_out.err`, in the directory where you started Spyder, for possible errors.

That's just a suggestion. If you want to phrase it differently, please go ahead.